File: IT/Containers/Esercizi/C++/esercizio11.cpp
/*
file esercizio11.cpp
compilare con: g++ -std=c++11 esercizio11.cpp
*/

#include <iostream>
#include <iterator>
#include <algorithm>
#include <vector>

using std::cout;
using std::endl;
using std::vector;
using std::ostream_iterator;
using std::boolalpha;
using std::sort;

int main()
{
	vector<int> a = {1, 2, 3};
	vector<int> b = {3, 2, 1};

	cout << "a = [";
	copy(a.begin(), a.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	cout << "b = [";
	copy(b.begin(), b.end(), ostream_iterator<int>(cout, ", "));
	cout << "\b\b]" << endl;

	vector<int> copy1 = a;
	vector<int> copy2 = b;

	sort(copy1.begin(), copy1.end());
	sort(copy2.begin(), copy2.end());

	cout << "a e b contengono gli stessi elementi? " << boolalpha << (copy1 == copy2) << endl;
}

File: IT/1/Containers/Esercizi/Java/Esercizio11.java
/* file Esercizio11.java */

import java.util.ArrayList;
import java.util.Collections;

public class Esercizio11 {
	public static void main(String [] args) {
		ArrayList<Integer> a = new ArrayList<Integer>();

		a.add(1);
		a.add(2);
		a.add(3);

		ArrayList<Integer> b = new ArrayList<Integer>();

		b.add(3);
		b.add(2);
		b.add(1);

		System.out.println("a = " + a);
		System.out.println("b = " + b);

		ArrayList<Integer> copy1 = new ArrayList<Integer>(a);
		ArrayList<Integer> copy2 = new ArrayList<Integer>(b);

		Collections.sort(copy1);
		Collections.sort(copy2);

		System.out.println("a e b contengono gli stessi elementi? " + copy1.equals(copy2));
	}
}

